src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -650,6 +651,22 @@ static int calibrate_mig(struct icap *icap)
 {
 	int i;
 
+	/* Check for any DDR or PLRAM banks that are in use */
+	bool is_memory_bank_connected = false;
+
+	/* If a DDR or PLRAM bank is found no need to keep searching */
+	for (i = 0; (i < icap->mem_topo->m_count) && (!is_memory_bank_connected); i++) {
+		struct mem_data* mem_bank = &icap->mem_topo->m_mem_data[i];
+		if (MEM_NEEDS_CALIBRATION(mem_bank->m_tag) && (mem_bank->m_used != 0))
```

Comment:
PLRAM is DRAM type. So I suggest to change as below:
if ((convert_mem_type(mem_topo->m_mem_data[i].m_tag) == MEM_DRAM ||
     mem_topo->m_mem_data[i].m_type == MEM_DRAM) && (mem_bank->m_used != 0))
@maxzhen Please comment if you have better thought.

Reply:
We should not use m_type from xclbin directly. It is not accurate and sometimes completely wrong. We should just use m_tag and convert it to type. So, I think we should extend convert_mem_type() and use it everywhere else for type conversion. Can we just convert "PLRAM" to DRAM?

Reply:
There are instances where PLRAM needs to be considered separate from DRAM so one function would not cut it. However, we could make a new function that returns an enumerated version of the tag. That would work very nicely. We end up with a new enum that describes the type of memory tag we have read.